python/rpdk/java/templates/init/guided_aws/StubUpdateHandler.java

```diff
@@ -134,7 +138,84 @@
                     })
                     .progress())
 
-            // STEP 4 [TODO: describe call/chain to return the resource model]
+            // If your resource supports tags, then the following pattern is required to handle stack level tags via soft-failing pattern
+            // STEP 4 [update stack level tags progress chain]
+            .then(progress -> {
+
+                // STEP 4.0 [initialize a proxy context]
+                // Stack level tag update should not force user but rather be optional, as it is possible that stack execution role will not have
+                // enough permissions to do so
+                // step in a discrete call/stabilize chain to ensure the entire resource is provisioned as intended.
+                ProgressEvent<ResourceModel, CallbackContext> event = proxy.initiate("{{ call_graph }}::{{ operation }}::stack-level-tags", proxyClient, progress.getResourceModel(), progress.getCallbackContext())
+
+                    // STEP 4.1 [TODO: construct a body of a request]
+                    .translateToServiceRequest((model) -> Translator.translateToStackTagUpdateRequest(request.getPreviousResourceTags(), request.getDesiredResourceTags()))
+
+                    // STEP 4.2 [TODO: make an api call]
+                    .makeServiceCall((awsRequest, client) -> {
+                        AwsResponse awsResponse = null;
+                        try {
+
+                            // TODO: put your post update resource code here
+
+                        } catch (final AwsServiceException e) {
+                            /*
+                            * While the handler contract states that the handler must always return a progress event,
+                            * you may throw any instance of BaseHandlerException, as the wrapper map it to a progress event.
+                            * Each BaseHandlerException maps to a specific error code, and you should map service exceptions as closely as possible
+                            * to more specific error codes
+                            */
+                            throw new CfnGeneralServiceException(ResourceModel.TYPE_NAME, e);
```

Comment:
Then why this is a soft-failing?

Reply:
stack level tags are not always intentional attachment by the customer so we dont really wanna enforce additional permissions when stack has not been changed on update